hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -1806,9 +1811,16 @@ func reconcileCAPIManagerDeployment(deployment *appsv1.Deployment, hc *hyperv1.H
 		},
 	}
 	hyperutil.SetColocation(hc, deployment)
-	hyperutil.SetRestartAnnotation(hc, deployment)
```

Comment:
I'm dropping this since I'm not sure there's a known reason to have it in a component different than the genuine control plane components. If we happen to see scalability issues not trivially solvable we can add it then. cc @relyt0925 wdyt?

Reply:
I think we need when a restart is requested to actually do a rolling restart of all components

Reply:
I do not think this should be removed. There's situations where things need to be restarted (like during ca rotation) and this is the current path to doing so (not all components auto reconcile the downstream ca certs, etc that they use)

Reply:
I'd prefer to do this when we see the need for it and we conclude there's no better automated way to solve it

Reply:
that's fine can we just comment it out with a todo for now to revaluate vs removing entirely